### PR TITLE
Sample ServiceEntries for apt-get, pip, and git tools showing how to grant access to mesh.

### PR DESCRIPTION
**Pull request by esnible**
Examples for accessing remote services.  See https://github.com/istio/istio/issues/11504 for motivation.

### CODE REVIEW

File: samples/external/aptget.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+metadata:
+  name: make-aptget-work
+spec:
+  hosts:
+  - deb.debian.org
+  - cdn-fastly.deb.debian.org
+  - security.debian.org
+  - archive.ubuntu.com
+  - security.ubuntu.com
+  ports:
```
Comment (sdake) [inline samples/external/aptget.yaml]: what about https when TLS is enabled in a Ubuntu container?

**Comment by esnible** [comment]
@frankbu Added README.  Let me know if you have any comments.

File: samples/external/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# External Services
+
+Istio-enabled services are unable to access services and URLs outside of the cluster. Pods use <i>iptables</i> to transparently redirect all outbound traffic to the sidecar proxy, which only handles intra-cluster destinations.
```
Comment (frankbu) [inline samples/external/README.md]: Should probably say "by default", esp with [this feature](https://deploy-preview-3140--preliminary-istio.netlify.com/docs/tasks/traffic-management/egress/#install-istio-with-access-to-all-external-services-by-default).

By default Istio-enabled services are unable ...

**Comment by ymesika** [comment]
cc @wenchenglu

File: samples/external/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# External Services
+
+By default
+Istio-enabled services are unable to access services and URLs outside of the cluster. Pods use <i>iptables</i> to transparently redirect all outbound traffic to the sidecar proxy, which only handles intra-cluster destinations.
```
Comment (venilnoronha) [inline samples/external/README.md]: nit: remove line break after "By default".

File: samples/external/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+See [the Egress Task](https://istio.io/docs/tasks/traffic-management/egress/) for
+information on Configuring Istio to contact external services.
```
Comment (venilnoronha) [inline samples/external/README.md]: s/Configuring/configuring

File: samples/external/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+This directory contains samples showing how to enable pods to contact a few well
+known services.
+
+If Istio is not configured to allow pods to contact external services the pods will
```
Comment (venilnoronha) [inline samples/external/README.md]: s/external services/external services,

File: samples/external/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+see errors such as 404s, HTTPS connection problems, and TCP connection problems.  If
+ServiceEntries are misconfigured pods may see problems with server names.
+
+## Try it out
+
+After an operator runs `kubectl create -f aptget.yaml` pods will be able to
+succeed with `apt-get update` and `apt-get install`.
+
+After an operator runs `kubectl create -f github.yaml` pods will be able to
+succeed with `git clone`.
```
Comment (venilnoronha) [inline samples/external/README.md]: nit: explicitly mention `git clone` on a GitHub repository.

**Pull request merged**